### PR TITLE
[Debug] Reintroduce ForYou pull-to-refresh error

### PR DESCRIPTION
**Pull request by damassi**
I've added logs so you can see this operate: 

screen shot 2017-12-21 at 12 07 35 pm

By adding in the ref check everything gets refreshed and no error, like it fixes a tiny race condition (somehow).

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Home/ArtistRails/__tests__/ArtistRail-tests.tsx`
- `src/lib/Scenes/Home/Components/ForYou/Components/__tests__/ArtworkCarousel-tests.tsx`
- `src/lib/Scenes/Home/Components/ForYou/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  [danger-plugin-jest] Could not read test results. Danger cannot pass or fail the build.
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Components/Home/ArtistRails/ArtistRail.tsx [hunk 1]
```diff
@@ -1,51 +1,64 @@
...
+  ScrollView,
+  StyleSheet,
+  TextStyle,
+  View,
+  ViewProperties,
+  ViewStyle
+} from "react-native";
 
-import metaphysics from "../../../metaphysics"
+import metaphysics from "../../../metaphysics";
```
Comment (damassi) [inline src/lib/Components/Home/ArtistRails/ArtistRail.tsx]: Hmm! i wonder whats going on with my editor, why its adding in semis? Maybe i need to update something....

Comment (damassi) [reply src/lib/Components/Home/ArtistRails/ArtistRail.tsx]: Is prettier configured correctly? Or was this deliberate?

Comment (orta) [reply src/lib/Components/Home/ArtistRails/ArtistRail.tsx]: Likely you needed  `yarn install` after pulling from master

Comment (damassi) [reply src/lib/Components/Home/ArtistRails/ArtistRail.tsx]: Ahh cool, I thought I did but will look into it 👍

Comment (orta) [reply src/lib/Components/Home/ArtistRails/ArtistRail.tsx]: If not that, try restarting VS Code, version of prettier may have changed under the hood and it's still referencing older settings

Comment (damassi) [reply src/lib/Components/Home/ArtistRails/ArtistRail.tsx]: Lol I am digging the 80 character column width tho 🐠  😄 But for sure, will fix

File: src/lib/Scenes/Home/Components/ForYou/Components/ArtworkCarousel.tsx [hunk 2]
```diff
@@ -230,36 +238,37 @@ export class ArtworkCarousel extends Component<Props & RelayPropsWorkaround, Sta
...
 
-  refreshData = () => {
+  refreshData = (railIndex = undefined) => {
     // TODO: Ensures rail has been mounted before a refresh occurs, circumventing setState errors.
     // See https://stackoverflow.com/a/40969739/1038901 for more info.
-    if (this.refs.rail) {
-      if (this.inflightRequest) {
-        this.inflightRequest.dispose()
-      }
+    // if (this.refs.rail) {
```
Comment (damassi) [inline src/lib/Scenes/Home/Components/ForYou/Components/ArtworkCarousel.tsx]: Note how i've now commented this out

**Comment by damassi** [comment]
@alloy - going to go ahead and close this since everything has been confirmed

**Pull request closed without merging**